src/main/java/com/englishtown/vertx/solr/impl/DefaultSolrService.java

```diff
@@ -54,11 +54,13 @@ public void query(VertxSolrQuery query, QueryOptions options, Handler<AsyncResul
         }
 
         try {
-            QueryRequest request = new QueryRequest(query);
```

Comment:
HI,

Documentation of SolrCloud says - SolrJ client class to communicate with SolrCloud. Instances of this class communicate with Zookeeper to discover Solr endpoints for SolrCloud collections, and then use the LBHttpSolrClient to issue requests.

So it internally creates URL for us as we are not passing any LBHttoSolrClient instance. correct me if I am wrong.

As you said , yes it works with same path. Zookeeper is used for cluster management for Solr data nodes for fault tolerance and high availability. 

Data node (shards) failure has no impact on the functionality with zookeeper. 

/select is used for http request to data nodes, so moved that code to DefaultVertxSolrClient.java

#1 - will extend the code to support customized solr options (connections etc.), for it works with default settings. not sure of customizing options at this time, still exploring solr.
#2 - will do that.
#3 - Sorry again. As i am relatively new to vertx , implementing piece by piece. will do that.
#4 - sure.

Please let me know if you have any questions. Thanks for review.